### DISCUSSION TITLE
Empty Manifest

### DISCUSSION THREAD

**Discussion by asbjornu** [Eptagone/Vite.AspNetCore#121] [Q&A]
Without setting `ViteOptions.Manifest`, it should be `.vite/manifest.json` according to the documentation. If I do the following in a controller, `True` is written to the console as expected:

```cs
public SomeController(IWebHostEnvironment environment)
{
    var manifestFile = environment.WebRootFileProvider.GetFileInfo(".vite/manifest.json");
    Console.WriteLine(manifestFile.Exists);
}
```

…so it's clear that the file `.vite/manifest.json` exists. However, when I inspect the `IViteManifest`, it returns no results:

<img width="2007" alt="Empty manifest" src="https://github.com/Eptagone/Vite.AspNetCore/assets/12283/6b89b2c4-acab-4b76-b329-69a7de1b4736">

How should I go about debugging this? It feels like the `ViteManifest` should not construct its own instance of `PhysicalFileProvider`, but instead use the one exposed on `environment`.

https://github.com/Eptagone/Vite.AspNetCore/blob/0c05a6b31823873a000cf8714e9a28a8bfaf8f04/src/Vite.AspNetCore/Services/ViteManifest.cs#L65

But I'm not sure if this would fix the problem or what the problem actually is. Thoughts?

**Comment by Eptagone** [comment]
Hello, did you set a subfolder as output folder? Example: `wwwroot/dist`? If that's the case, then you need to configure the [base](https://github.com/Eptagone/Vite.AspNetCore?tab=readme-ov-file#available-options) option or the package won't be able to find your manifest file. See more in the [Wiki](https://github.com/Eptagone/Vite.AspNetCore/wiki#how-to-configure-a-subfolder-as-output-for-my-vite-assets).
